### DISCUSSION TITLE
请问一下怎么连接MongoDB或者mysql数据库？

### DISCUSSION THREAD

**Discussion by fengjiajun169** [ddiu8081/chatgpt-demo#315] [Q&A] [answered]
1.现在使用的是edge，edge模式下我连接数据库，在编译的时候会报错，提示很多依赖找不到，看Astro的官方文档说是大部分Node.js的api不支持；
2. 如果使用serverless，可以使用数据库，但是输入promot后生成答案会慢一点，而且答案是一起出来的，不是流式的效果了

折腾了好久没折腾好，有大佬知道吗？

**Comment by yzh990918** [comment, marked as answer]
可以用在 Node18 环境下试试, edge serverless不是必要条件
